updating/updating-disconnected-cluster.adoc

```diff
@@ -1,23 +1,134 @@
+
+.Procedure
+
+. Findout if there is link:https://access.redhat.com/solutions/4583231[an update path] exists between your current version and the version you want the cluster to move.
+
+. Mirror the version images to the internal container registry.
+
+** Connect the removable media to a system connected to the internet.
+** Mirror the images and configuration manifests to a directory on the removable media.
++
+----
+$ oc adm release mirror "--to-dir=${REMOVABLE_MEDIA_PATH}/mirror" quay.io/openshift-release-dev/ocp-release:4.3.0-x86_64
+----
++
+
+** Take the media to the disconnected environment and upload the images to the local container registry
++
+----
+$ oc image mirror "--from-dir=${REMOVABLE_MEDIA_PATH}/mirror" file://openshift/release:4.3.0* REGISTRY/REPOSITORY
+----
++
+. Use `oc` to log in to a given cluster to be upgraded.
++
+[NOTE]
+====
+If you are upgrading from earlier release than 4.5.0, you must must use the manual method of creating configmap.
+
+This is because `oc` command enhancement is not available in previous releases.
+====
++
+. Apply the the mirrored release image signature ConfigMap to the connected cluster
++
+----
+$ oc apply -f mirror/config/signature-sha256-81154f5c03294534.yaml
+----
+. Perform the upgrade
++
+----
+$ oc adm upgrade --allow-explicit-upgrade --to-image REGISTRY/REPOSITORY@sha256:81154f5c03294534e1eaf0319bef7a601134f891689ccede5d705ef659aa8c92
+----
+
+[id="update-disconnected-manual_method_configmap_creation"]
+== Manual method to create the image signature configmap
+
+The following commands will create and apply the image signature configmap to the cluster you want to update. 
+
+.Procedure
+
+. Find the link:https://access.redhat.com/solutions/4583231[version] you want the cluster to update.
+
+. Add the version to OCP_RELEASE_NUMBER environment variable.
++
+----
+$ export OCP_RELEASE_NUMBER=<4.y.z>
+----
++
+. Get the release image digest from link:https://quay.io/[Quay].
++
+----
+$ export DIGEST="$(oc adm release info quay.io/openshift-release-dev/ocp-release:${OCP_RELEASE_NUMBER}-x86_64 | sed -n 's/Pull From: .*@//p')"
+----
++
+. Get the image signature from link:https://mirror.openshift.com/pub/openshift-v4/signatures/openshift/release[mirror.openshift.com] website.
++
+----
+$ export SIGNATURE_BASE64=$(curl -s "https://mirror.openshift.com/pub/openshift-v4/signatures/openshift/release/$(echo $DIGEST | cut -d: -f1)=$(echo $DIGEST | cut -d: -f2)/signature-1" | base64 -w0 && echo)
+----
++
+. Get the digest algorithm.
++
+----
+export DIGEST_ALGO=$(echo $DIGEST | cut -d: -f1)
+----
++
+. Get the digest signature.
++
+----
+export DIGEST_SIGNATURE=$(echo $DIGEST | cut -d: -f2)
+----
++
+. Create the configmap.
++
+----
+$ cat >checksum-${OCP_RELEASE_NUMBER}.yaml <<EOF
```

Comment:
Probably better to use:

```console
$ DIGEST=sha256=a3fe9de9b338abc80e2afafdf38dff0c2de3efb61c6896e8e16495f59e717f53
$ DIGEST_ALGO="${DIGEST%%:*}"
$ DIGEST_ENCODED="${DIGEST#*:}"
$ curl -s "https://mirror.openshift.com/pub/openshift-v4/signatures/openshift/release/${DIGEST_ALGO}=${DIGEST_ENCODED}/signature-1" >"${DIGEST_ALGO}-${DIGEST_ENCODED}-1"
$ oc -n openshift-config-managed create configmap -o yaml --dry-run --from-file="${DIGEST_ALGO}-${DIGEST_ENCODED}-1" "release-image-signature-${OCP_RELEASE_NUMBER}"
```

Not sure if there's a convenient way to stuff in the `release.openshift.io/verification-signatures` label though...

Reply:
@wking I do not understand that point. You need to be more verbose :)

Reply:
You need [the label][1] you were setting in the here-doc.  I could not find a way to get `oc create configmap` to set labels.  Maybe I missed it?  Maybe we can add the label in post-processing with `-o json` and `jq`.  Maybe we want the here-doc and manual base64ing after all?

[1]: https://github.com/openshift/oc/blob/8dda2e784003f0bd16142a3c4a63924768bdd891/pkg/helpers/release/configmap.go#L28-L32

Reply:
I reverted to here-doc.